mlflow/projects/__init__.py

```diff
@@ -280,6 +300,23 @@ def _fetch_project(uri, force_tempdir, version=None, git_username=None, git_pass
     return res
 
 
+def _unzip_repo(uri, dst_dir):
+    import zipfile
+    with zipfile.ZipFile(uri) as zip_file:
+        zip_file.extractall(dst_dir)
+
+
+def _fetch_zip_repo(uri, dst_dir):
+    import requests
+    import zipfile
+    from io import BytesIO
+    response = requests.get(uri)
+    if requests.codes.ok != response.status_code:
+        raise ExecutionException("Unable to retrieve ZIP file %s" % uri)
+    with zipfile.ZipFile(BytesIO(response.content)) as zip_file:
```

Comment:
Let's modify this function to return the `BytesIO` stream directly. We can then invoke `_unzip_repo` on the stream to avoid duplicating the `zipfile` functions.